ChefExtensionHandler/bin/chef-install.psm1

```diff
@@ -51,6 +51,28 @@ function Get-LocalDestinationMsiPath {
   [System.IO.Path]::GetFullPath("$env:temp\\chef-client-latest.msi")
 }
 
+function Get-Settings-File {
+  $configRootPath = Chef-GetExtensionRoot + "\\RuntimeSettings"
+  $configFilesPath = "$configRootPath\\*.settings"
+  $configFile = ls $configFilesPath | Sort-Object { [Int] ($_.basename -Replace '\D') } | Select -Last 1 | Select -ExpandProperty Name
+  if (!$configFile)
+  {
+    Write-Host("[$(Get-Date)] No config file found !!")
+    exit 1
+  }
+  $configFilePath = $configRootPath + "\\" + $configFile
+  
+  $configFilePath
+}
+
+function Get-Chef-Version {
+  $settingsFile = Get-Settings-File
+  $settingsData = Get-Content $settingsFile -Raw | ConvertFrom-Json
```

Comment:
I am doing this operation before chef-client download and installation begins, so I guess the above Ruby function won't work as Ruby won't be available on the target node until the installation of chef-client completes.